central/processbaseline/evaluator/evaluator_impl.go

```diff
@@ -86,29 +87,29 @@ func (e *evaluator) EvaluateBaselinesAndPersistResult(deployment *storage.Deploy
 		}
 	}
 
-	var processes []*storage.ProcessIndicator
+	var processes []*views.ProcessIndicatorRiskView
 	if hasAtLeastOneLockedBaseline {
-		processes, err = e.indicators.SearchRawProcessIndicators(evaluatorCtx, search.NewQueryBuilder().AddExactMatches(search.DeploymentID, deployment.GetId()).ProtoQuery())
+		processes, err = e.indicators.GetProcessIndicatorsRiskView(evaluatorCtx, search.NewQueryBuilder().AddExactMatches(search.DeploymentID, deployment.GetId()).ProtoQuery())
```

Comment:
Since we do not need the slice and we iterate over it I think here we should have a callback approach to not allocate the full slice.

Reply:
I generally agree.  Any objection to doing that as follow up?  I think what would be nice to do is update (or create a new function) `RunSelectRequestForSchema` to work more like `RunQueryForSchemaFn`